python-packages/smithy-http/tests/unit/test_fields.py

```diff
@@ -222,3 +222,17 @@ def test_fields_length_value(fields: Fields, expected_length: int) -> None:
 )
 def test_fields_repr(fields: Field, expected_repr: str) -> None:
     assert repr(fields) == expected_repr
+
+
+@pytest.mark.parametrize(
+    "fields,key,contained",
+    [
+        (Fields(), "bad_key", False),
+        (Fields([Field(name="fname1")]), "fname1", True),
+        (Fields([Field(name="fname2")]), "fname1", False),
+        (Fields([Field(name="f1"), Field(name="f2")]), "f1", True),
+        (Fields([Field(name="f1"), Field(name="f2")]), "f3", False),
```

Comment:
If we decided to normalize the key parameter, we should add a test with an uppercase input key like below:
```
(Fields([Field(name="f1"), Field(name="f2")]), "F1", False),
```
If not, ignore this comment.

Reply:
Yep good call out!